src/process_utility.c

```diff
@@ -1428,6 +1427,37 @@ process_altertable_chunk(Hypertable *ht, Oid chunk_relid, void *arg)
 	AlterTableInternal(chunk_relid, list_make1(cmd), false);
 }
 
+static void
+process_altertable_set_tablespace_end(Hypertable *ht, AlterTableCmd *cmd)
+{
+	Oid			tspc_oid = get_rel_tablespace(ht->main_table_relid);
+	NameData	tspc_name;
+	Tablespaces *tspcs;
+
+	Assert(OidIsValid(tspc_oid));
+	namestrcpy(&tspc_name, cmd->name);
+
+	tspcs = tablespace_scan(ht->fd.id);
+
+
+	if (tspcs->num_tablespaces > 1)
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
```

Comment:
I am wondering if this is the appropriate error code here. We support the feature, but just not the current state.

Perhaps `ERRCODE_OBJECT_NOT_IN_PREREQUISITE_STATE` is more appropriate or this could also be a case where we should expand our set of own error codes?